hane-syntax/src/lower.rs

```diff
@@ -63,6 +80,60 @@ impl Command {
                 let ttype = ttype.lower(global, &mut names)?;
                 global.insert(name.clone());
                 LoweredCommandVariant::Axiom(name, ttype)
+            },
+            CommandVariant::Inductive(mut bodies) => {
+                let params = std::mem::take(&mut bodies[0].params);
+                bodies.iter()
+                    .skip(1)
+                    .map(|body|&body.params)
+                    .try_for_each(|ps|
+                        if &params == ps {
+                            Ok(())
+                        } else {
+                            Err(SpanError { span: self.span.clone(), err: LoweringError::ParamsMustMatch })
+                        }
+                    )?;
+
+                let mut lowered_params = Vec::with_capacity(params.len());
+                for param in params {
+                    lowered_params.push((param.name.clone(), param.ttype.lower(global, &mut names)?));
+                    names.push(param.name);
```

Comment:
Shouldn't these be cleaned up on error?

Reply:
Both `lowered_params` and `names` are local variables. So there's no need.